tests/test_utils.py

```diff
@@ -117,6 +121,45 @@ def test_getExpPositionOffset(self):
             self.assertAlmostEqual(result.deltaDec.asDegrees(), deltaDec, 6)
 
 
+class ImageBasedTestCase(lsst.utils.tests.TestCase):
+    """A test case for testing sky position offsets for exposures."""
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.butler = makeDefaultLatissButler()
+        except FileNotFoundError:
+            raise unittest.SkipTest("Skipping tests that require the LATISS butler repo.")
+
+        cls.dataId = {'day_obs': 20200315, 'seq_num': 120, 'detector': 0}
+        cls.bestEffort = BestEffortIsr()  # should always succeed if makeDefaultLatissButler works
+        cls.exp = cls.bestEffort.getExposure(cls.dataId)
+
+    def test_fluxFromFootprint(self):
+        footPrintSet = detectObjectsInExp(self.exp)
+        allFootprints = footPrintSet.getFootprints()
+        self.assertGreaterEqual(len(allFootprints), 3)
+        singleFootprint = allFootprints[0]
+        twoFootprints = allFootprints[0:2]
+
+        # check it can accept a footprintSet, and single and iterables of
+        # footprints
+        for obj in (footPrintSet, singleFootprint, twoFootprints):
```

Comment:
I'd rather this not be in a loop: you have to do an `isinstance` as part of the test anyway. Pulling it out into individual blocks (possibly with a helper `_checkFluxFromFootprints` method) will make it easier to tell where a failure is.